go-controller/pkg/node/gateway_shared_intf.go

```diff
@@ -1384,7 +1385,39 @@ func flowsForDefaultBridge(bridge *bridgeConfiguration, extraIPs []net.IP) ([]st
 			fmt.Sprintf("cookie=%s, priority=10, table=1, dl_dst=%s, actions=output:%s",
 				defaultOpenFlowCookie, bridgeMacAddress, ofPortHost))
 	}
+
 	defaultNetConfig := bridge.netConfig[types.DefaultNetworkName]
+
+	for _, svcIP := range udnAllowedServicesIPs {
+		ipPrefix := "ip"
+		masqueradeSubnet := config.Gateway.V4MasqueradeSubnet
+		if !utilnet.IsIPv4(svcIP) {
+			ipPrefix = "ipv6"
+			masqueradeSubnet = config.Gateway.V6MasqueradeSubnet
+		}
+
+		// The flows added below have a higher priority than the default network service flows:
+		//   priority=500,ip,in_port=LOCAL,nw_dst=10.96.0.0/16 actions=ct(commit,table=2,zone=64001,nat(src=169.254.0.2))
+		//   priority=500,ip,in_port="patch-breth0_ov",nw_src=10.96.0.0/16,nw_dst=169.254.0.2 actions=ct(table=3,zone=64001,nat)
+		// This ordering ensures that there is no SNAT for UDN originated traffic.
+		// However, it also means that the flows match traffic with the source IP of the default masquerade IP (169.254.0.2) and a
+		// destination of a UDN allowed service. To handle this, the flow commits the connection to conntrack and applies NAT
+		// action in the reply direction to potentially unSNAT from 169.254.0.2.
+
+		// table 0, user-defined network host -> OVN towards default cluster network services
```

Comment:
In general it's usually a bad idea to only commit some of the traffic.  If the traffic that is not committed to conntrack happens to have a 4-tuple that also matches what we get post-nat for subsequent traffic that we nat and commit, netfilter has no way of figuring that out and won't do any tuple conflict resolution.

Reply:
it will help us debug if it is conntrack, i was just trying to do it as a perf optimization. Since we only send CT traffic that matches on 169.254.0.2 I thought it would be fine, but we can commit everything for now.

Reply:
Thanks! I guess lets keep it for now and I can change the approach if needed in the future.